### PR TITLE
Feature/add generator for submodels

### PR DESCRIPTION
**Pull request by andreas-schilling**
Implement https://github.com/eclipse-esmf/esmf-sdk/issues/280

### CODE REVIEW

File: core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/AspectModelAASGenerator.java [hunk 1]
```diff
@@ -55,28 +65,71 @@ public void generateAasXmlFile(
...
+         output.write( generateXmlOutput( Map.of( aspect, aspectData ) ).toByteArray() );
+      }
+   }
+
+   protected ByteArrayOutputStream generateXmlOutput( final Map<Aspect, JsonNode> aspectsWithData ) throws IOException {
+      final AspectModelAASVisitor visitor = new AspectModelAASVisitor().withPropertyMapper( new LangStringPropertyMapper() );
+
+      final Map<Aspect, Environment> aspectEnvironments =
+            aspectsWithData.entrySet().stream()
+                           .map( aspectWithData -> {
```
Comment (atextor) [inline core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/AspectModelAASGenerator.java]: Indentation looks wrong here, the dots should not be aligned. Please check/adjust format across all files.

File: core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/DefaultPropertyMapper.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.eclipse.digitaltwin.aas4j.v3.model.impl.DefaultProperty;
+
+import io.openmanufacturing.sds.metamodel.Type;
+
+/**
+ * The default mapper used for all properties.
+ */
+public class DefaultPropertyMapper implements PropertyMapper<Property> {
+   @Override
+   public Property mapToAasProperty( Type type, io.openmanufacturing.sds.metamodel.Property property, Context context ) {
```
Comment (atextor) [inline core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/DefaultPropertyMapper.java]: Please adjust Save Actions; arguments should be final

**Review by atextor — CHANGES_REQUESTED** [review]

File: core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/AspectModelAASGenerator.java (hunk 1 above)
Comment (andreas-schilling) [reply core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/AspectModelAASGenerator.java]: done.

File: core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/DefaultPropertyMapper.java (hunk 2 above)
Comment (andreas-schilling) [reply core/sds-aspect-model-aas-generator/src/main/java/io/openmanufacturing/sds/aspectmodel/aas/DefaultPropertyMapper.java]: done.

**Review by atextor — APPROVED** [review]

**Review by atextor — APPROVED** [review]

**Review by atextor — APPROVED** [review]

**Comment by andreas-schilling** [comment]
Created new PR that directly targets the upstream repo. See https://github.com/eclipse-esmf/esmf-sdk/pull/364

**Pull request closed without merging**